### PR TITLE
Change deployment scope for MG Diagnostics #338

### PR DESCRIPTION
**Pull request by lachaves**
# Overview/Summary

This PR changes the scope of the orchestration module mgDiagSettingsAll from tenant to mg level. 

### Breaking Changes

None

## Testing Evidence

The validation available here https://learn.microsoft.com/en-us/rest/api/monitor/management-group-diagnostic-settings/get?tabs=HTTP&tryIt=true&source=docs#code-try-0 was used to test. 

## As part of this Pull Request I have

- [X] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [X] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [ X] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ X] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [X] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [X ] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [X] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)
## As part of this Pull Request I have

- [X] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [X] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [ X] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ X] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [X] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [X ] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [X] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

File: infra-as-code/bicep/orchestration/mgDiagSettingsAll/mgDiagSettingsAll.bicep [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+targetScope = 'managementGroup'
 
-@description('Prefix used for the management group hierarchy in the managementGroups module. Default: alz')
+@description('Prefix used for the management group hierarchy in the managementGroups module.')
 @minLength(2)
 @maxLength(10)
 param parTopLevelManagementGroupPrefix string = 'alz'
 
-@description('Dictionary Object to allow additional or different child Management Groups of the Landing Zones Management Group. Default: Empty Array')
+@description('Dictionary Object to allow additional or different child Management Groups of the Landing Zones Management Group .')
```
Comment (jtracey93) [inline infra-as-code/bicep/orchestration/mgDiagSettingsAll/mgDiagSettingsAll.bicep:8]:
```suggestion
@description('Dictionary Object to allow additional or different child Management Groups of the Landing Zones Management Group.')
```

File: infra-as-code/bicep/orchestration/mgDiagSettingsAll/mgDiagSettingsAll.bicep [hunk 2]
```diff
@@ -84,5 +84,6 @@ module modMgChildrenDiagSet '../../modules/mgDiagSettings/mgDiagSettings.bicep'
 module modCustomerUsageAttribution '../../CRML/customerUsageAttribution/cuaIdTenant.bicep' = if (!parTelemetryOptOut) {
   #disable-next-line no-loc-expr-outside-params //Only to ensure telemetry data is stored in same location as deployment. See https://github.com/Azure/ALZ-Bicep/wiki/FAQ#why-are-some-linter-rules-disabled-via-the-disable-next-line-bicep-function for more information //Only to ensure telemetry data is stored in same location as deployment. See https://github.com/Azure/ALZ-Bicep/wiki/FAQ#why-are-some-linter-rules-disabled-via-the-disable-next-line-bicep-function for more information
   name: 'pid-${varCuaid}-${uniqueString(deployment().location)}'
+  scope: tenant()
   params: {}
 }
```
Comment (jtracey93) [inline infra-as-code/bicep/orchestration/mgDiagSettingsAll/mgDiagSettingsAll.bicep:89]: Change this module out to use the management group scoped level version here https://github.com/Azure/ALZ-Bicep/blob/main/infra-as-code/bicep/CRML/customerUsageAttribution/cuaIdManagementGroup.bicep

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks @lachaves for your work here. A couple of small suggestions plus the below:

Make updates to the tests here:

1. https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml#L108
2. https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml#L197
3. https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml#L88

Thanks and let me know when good to review again 👍

**Comment by lachaves** [comment]
@jtracey93, Thanks for your review, suggestions and updates completed. Please validate and let me know if anything still missing.

**Review by jtracey93 — APPROVED** [review]
LGTM

File: infra-as-code/bicep/orchestration/mgDiagSettingsAll/README.md [hunk 3]
```diff
@@ -50,10 +50,14 @@ Below are some examples of how to use this input parameter in both Bicep & JSON
 ##### Bicep Example
 
 ```bicep
-parLandingZoneMgChildren: [
-    'pci'
-    'another-example'
-]
+parLandingZoneMgChildren: {
```
[the remark this answers is not in the document]
Comment (jtracey93) [reply infra-as-code/bicep/orchestration/mgDiagSettingsAll/README.md:53]: @lachaves to comment

[the remark this answers is not in the document]
Comment (lachaves) [reply infra-as-code/bicep/orchestration/mgDiagSettingsAll/README.md:53]: I have changed the description in the parameters table and also the bicep example for the parameter to show as an array type.  Sending a PR with updates.

**Pull request merged**